cluster-autoscaler/cloudprovider/openshiftmachineapi/machineapi_controller.go

```diff
@@ -443,7 +443,9 @@ func (c *machineController) machineSetNodeGroups() ([]*nodegroup, error) {
 		if err != nil {
 			return err
 		}
-		if ng.MaxSize()-ng.MinSize() > 0 && pointer.Int32PtrDerefOr(machineSet.Spec.Replicas, 0) > 0 {
+		if ng.scalableResource.CanScaleFromZero() {
+			nodegroups = append(nodegroups, ng)
+		} else if ng.MaxSize()-ng.MinSize() > 0 && pointer.Int32PtrDerefOr(machineSet.Spec.Replicas, 0) > 0 {
```

Comment:
shouldn't the check `ng.MaxSize()-ng.MinSize() > 0` be at the top?
So no matter this node group can scale from zero or not. If ``ng.MaxSize()-ng.MinSize() > 0` is false it should not be included.

Reply:
i.e.
```
if ng.MaxSize()-ng.MinSize() > 0
    if ng.scalableResource.CanScaleFromZero()
             append
    else if pointer.Int32PtrDerefOr(machineSet.Spec.Replicas, 0) > 0
             append
```

Reply:
The same logic we settle here needs to be included for `machineDeploymentNodeGroups`

Reply:
this logic makes sense to me, if essentially if MaxSize < MinSize then we have bigger problems.

i like your solution of gating on MaxSize - MinSize > 0 first, i will change this code to follow that.